### PR TITLE
fix(flow): batch builder with type

### PR DESCRIPTION
**Pull request by discord9**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

fix a bug in which null type vector builder is used incorrectly

Please explain IN DETAIL what the changes are in this PR and why they are needed:

- previously, some vector builder is used wrong and could be null vector builder when encounter null value, this is fixed now as passing actual type is supported to build vector&batch in flow

## Checklist

- [ ] I have written the necessary rustdoc comments.
- [ ] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.

### CODE REVIEW

**Comment by discord9** [comment]
Fixing failing tests now

File: src/flow/src/expr.rs [hunk 1]
```diff
@@ -221,10 +223,28 @@ impl Batch {
...
+                } else {
+                    // both are null, so we will push null type
+                    dts.push(datatypes::prelude::ConcreteDataType::null_datatype())
+                }
+            }
+            if self.batch.is_empty() {
+                other.batch.iter().map(|v| v.data_type()).collect_vec()
+            } else {
+                self.batch.iter().map(|v| v.data_type()).collect_vec()
+            }
```
Comment (evenyag) [inline src/flow/src/expr.rs:247]: Looks like we need to remove them.

File: src/flow/src/expr.rs [hunk 2]
```diff
@@ -221,10 +223,28 @@ impl Batch {
...
+                    dts.push(v.data_type())
+                } else if let Some(v) = other.batch().get(i)
+                    && !v.data_type().is_null()
+                {
+                    dts.push(v.data_type())
+                } else {
+                    // both are null, so we will push null type
+                    dts.push(datatypes::prelude::ConcreteDataType::null_datatype())
+                }
+            }
```
Comment (evenyag) [inline src/flow/src/expr.rs:242]: We never read this `dts` outside of the block.

File: src/flow/src/adapter.rs [hunk 3]
```diff
@@ -555,6 +555,8 @@ impl FlowWorkerManager {
         let default_interval = Duration::from_secs(1);
         let mut avg_spd = 0; // rows/sec
         let mut since_last_run = tokio::time::Instant::now();
+        let run_per_trace = 5;
```
Comment (evenyag) [inline src/flow/src/adapter.rs:558]: I wonder if `5` is too small.

File: src/flow/src/compute/render/reduce.rs [hunk 4]
```diff
@@ -47,6 +47,7 @@ impl Context<'_, '_> {
         reduce_plan: &ReducePlan,
         output_type: &RelationType,
     ) -> Result<CollectionBundle<Batch>, Error> {
+        common_telemetry::debug!("render reduce batch");
```
Comment (evenyag) [inline src/flow/src/compute/render/reduce.rs:50]: Do we still need this log?

**Review by fengjiachun — APPROVED** [review]
LGTM

**Comment by fengjiachun** [comment]
@discord9 Some conflits

**Pull request merged**